service/controller/app/key/key_test.go

```diff
@@ -11,44 +12,82 @@ import (
 )
 
 func Test_AppConfigMapName(t *testing.T) {
-	expectedName := "giant-swarm-configmap-name"
-
-	obj := v1alpha1.App{
-		Spec: v1alpha1.AppSpec{
-			Name:    "giant-swarm-name",
-			Catalog: "giant-swarm-catalog-name",
-			Config: v1alpha1.AppSpecConfig{
-				ConfigMap: v1alpha1.AppSpecConfigConfigMap{
-					Name:      "giant-swarm-configmap-name",
-					Namespace: "giant-swarm-configmap-namespace",
+	testCases := []struct {
+		name          string
+		obj           v1alpha1.App
+		expectedValue string
+	}{
```

Comment:
I checked https://github.com/giantswarm/apiextensions/pull/285 and it's pretty old. So we were not using kubebuilder and the optional fields were not set to nullable.

So no changes are required but I extended the unit test coverage. As if we change apiextensions in future and break this we will have a lot of problems.